src/dmd/denum.d

```diff
@@ -308,7 +308,10 @@ extern (C++) final class EnumDeclaration : ScopeDsymbol
                 return memtype.defaultInit(loc);
             }
 
-            error(loc, "forward reference of `%s.init`", toChars());
+            if (semanticRun >= PASS.semanticdone)
+                error(loc, "is opaque and has no default initializer");
+            else
+                error(loc, "forward reference of `%s.init`", toChars());
```

Comment:
This line isn't covered because actual forward references crash dmd...

Will open another PR for those errors.